vars/komodo.groovy

```diff
@@ -1,255 +1,175 @@
-                    }
-                }
-                steps {
-                    script {
-                        installLocalFiles(env.KOMODO_RELEASES_ROOT, env.PREFIX, env.RELEASE_NAME, env.PERMISSIONS_EXEC)
+
+                    dir(getReleaseName()) {
+                        script {
+                            if (fileExists("."))
+                                error "Build ${getReleaseName()} is already in progress."
+                        }
+
+                        gitClone "https://github.com/${params.CODE_GIT_FORK}/komodo.git",
+                            params.CODE_GIT_REF
+                        gitClone "https://github.com/${params.CONFIG_GIT_FORK}/komodo-releases.git",
+                            params.CONFIG_GIT_REF
+
+                        script {
+                            def komodoPath = "${env.WORKSPACE}/${getReleaseName()}/komodo"
+                            def configPath = "${env.WORKSPACE}/${getReleaseName()}/komodo-releases"
+                            def releaseFile = "${getReleaseName()}.yml"
+                            def matrixFile = "${configPath}/releases/matrices/${params.MATRIX_FILE_BASE}.yml"
+
+                            // Bootstrap
+                            sh "cd ${komodoPath}; ./bootstrap.sh ${getPython()}"
+
+                            // Transpile
+                            sh "cd ${configPath}; ${komodoPath}/boot/kmd-env/bin/komodo-transpiler transpile --matrix-file=${matrixFile} --output releases"
+
+                            // Lint
+                            sh "cd ${configPath}; ${komodoPath}/boot/kmd-env/bin/komodo-lint releases/${releaseFile} repository.yml"
+
+                            // Run!
+                            withEnv(["TMPDIR=${configPath}/tmp"]) {
+                                sh """
+                            cd ${configPath}
+                            ${komodoPath}/runkmd.sh                                    \
+                                releases/${releaseFile}                                \
+                                repository.yml                                         \
+                                --download                                             \
+                                --build                                                \
+                                ${params.deploy ? '--install' : '--dry-run'}           \
+                                --jobs 6                                               \
+                                --release ${getReleaseName()}                          \
+                                --tmp tmp                                              \
+                                --cache cache                                          \
+                                --prefix ${params.PREFIX}                              \
+                                --pip ${komodoPath}/boot/build-env/bin/pip             \
```

Comment:
Unnecessary. The correct `pip` is already in `PATH` via `runkmd.sh`. Same should be true for `virtualenv`.

Reply:
Unfortunately, some packages will require that `PATH` be set to the builds's `bin` path. For builtin python, that `pip` is going to be old and quite possibly not work at all.

Reply:
Actually, I think it's necessary. I will demonstrate.